code/modules/virology/virology_goals.dm

```diff
@@ -0,0 +1,178 @@
+			if("stage rate")
+				goal_property_value += S.stage_speed
+			if("transmittable")
+				goal_property_value += S.transmittable
+		qdel(S)
+	while(check_for_duplicate())
+
+/datum/virology_goal/property_symptom/check_for_duplicate()
+	. = FALSE
+	if(!goal_symptom || !goal_property || !goal_property_value)
+		return
+	for(var/datum/virology_goal/property_symptom/V in (GLOB.archived_virology_goals + GLOB.virology_goals))
+		if(goal_symptom == V.goal_symptom && goal_property == V.goal_property && goal_property_value == V.goal_property_value)
+			return TRUE
+
+/datum/virology_goal/property_symptom/get_report(is_ui = FALSE)
+	var/return_text
+	if(!is_ui)
+		return_text += "<b>Effects of [goal_symptom_name] symptom and level [goal_property_value] [goal_property]</b><br>"
+	return_text += "Viral samples with a specific symptom and properties are required to study the effects of this symptom in various conditions. We need you to deliver [delivery_goal]u of viral samples containing the [goal_symptom_name] symptom and with the [goal_property] property at level [goal_property_value] along with 3 other symptoms to us through the cargo shuttle."
+	return return_text
+
+/datum/virology_goal/property_symptom/check_completion(list/datum/reagent/reagent_list)
+	. = FALSE
+	var/datum/reagent/blood/BL = locate() in reagent_list
+	if(!BL)
+		return
+	if(BL.data && BL.data["viruses"])
+		for(var/datum/disease/advance/D in BL.data["viruses"])
+			if(length(D.symptoms) < 4) //We want 3 other symptoms alongside the requested one
+				continue
+			var/properties = D.GenerateProperties()
+			var/property = properties[goal_property]
+			if(property != goal_property_value)
+				continue
+			for(var/datum/symptom/S in D.symptoms)
+				if(!goal_symptom)
+					return
+				if(S.type != goal_symptom)
+					continue
+				delivered_amount += BL.volume
+				if(delivered_amount >= delivery_goal)
+					completed = TRUE
+					check_total_virology_goals_completion()
+					return TRUE
+
+/datum/virology_goal/virus
+	name = "Specific Viral Sample Request (Non-Stealth)"
+	var/list/goal_symptoms = list() //List of type paths of the symptoms, we could go with a diseaseID here instead a list of symptoms but we need the list to tell the player what symptoms to include
+
+/datum/virology_goal/virus/New()
+	do
+		goal_symptoms = list()
+		var/list/symptoms = subtypesof(/datum/symptom)
+		var/stealth = 0
+		for(var/i in 1 to 5)
+			var/list/candidates = list()
+			for(var/V as anything in symptoms) //There is a "as anything" added because for some mystery reason it doesnt work without it
+				var/datum/symptom/S = V
+				if(istype(src, /datum/virology_goal/virus/stealth))
+					if(stealth + S.stealth < 3) //The Pandemic cant detect a virus with stealth 3 or higher and we want that, this is a stealth virus
+						continue
+				else
+					if(stealth + S.stealth >= 3) //The Pandemic cant detect a virus with stealth 3 or higher and we dont want that, this isnt a stealth virus
+						continue
```

Comment:
Instead of doing an `istype(src)`, please make a function called something like `meets_stealth_requirement` that takes an stealth parameter (which you pass in as stealth + S.stealth), and returns `stealth < 3`. And then for the stealth goal, you override that proc to be `stealth > 3`
```suggestion
				if(!meets_stealth_requirement(stealth + S.stealth)
					continue
```
```
/datum/virology_goal/virus/proc/meets_stealth_requirement(stealth)
	return (stealth < 3)
	
/datum/virology_goal/virus/stealth/meets_stealth_requirement(stealth)
	return (stealth >= 3)